balancer/balancer.go

```diff
@@ -110,6 +110,11 @@ type SubConn interface {
 	UpdateAddresses([]resolver.Address)
 	// Connect starts the connecting for this SubConn.
 	Connect()
+	// GetOrBuildProducer returns a reference to the existing Producer for this
+	// ProducerBuilder in this SubConn, or, if one does not currently exist,
+	// creates a new one and returns it.  Returns a close function which must
+	// be called when the Producer is no longer needed.
+	GetOrBuildProducer(ProducerBuilder) (p Producer, close func())
```

Comment:
Was the option of adding a `Close()` method on the `Producer` considered instead of returning a close func?

Also, should we return an error from `GetOrBuildProducer` or is the error condition represented by a `nil` `Producer`. If so, could we document that?

Reply:
Yes.  Per offline explanation, the `close` method is necessary to prevent inappropriate calling while also allowing for a type assertion of the returned `Producer` directly to the type the `ProducerBuilder` creates.

Reply:
This isn't possible; we will always return a non-nil `Producer`.